gist/src/main/java/build/gist/data/listeners/Queue.kt

```diff
@@ -124,6 +134,15 @@ class Queue: GistListener {
         }
     }
 
+    private fun addMessageToLocalStore(message: Message) {
+        val localMessage = localMessageStore.find { localMessage -> localMessage.queueId == message.queueId }
+        if (localMessage == null) { localMessageStore.add(message) }
```

Comment:
Since `localMessageStore` must contain unique `Message`s with no duplicates, what about using a `Set`? I think that would make the add and remove functions in this class more concise.

Reply:
Would second this, `Set` might make more sense here. 

*Note:* If you do make it `set`, use a custom `equals` implementation or a custom `hashCode` implementation to define a custom equality for elements in a set, which then determines the uniqueness of elements in the set.

Reply:
The issue with using set is that when messages are fetched remotely their `instanceId` is generated on instantiation so we have to make a comparison by QueueId only.

Reply:
I think you can make comparisons on the `instanceId` field, even though it is generated during instantiation. This should work, 

Inside Message class
```
override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other !is Message) return false

    return messageId == other.messageId &&
           instanceId == other.instanceId &&
           queueId == other.queueId
}

override fun hashCode(): Int {
    var result = messageId.hashCode()
    result = 31 * result + instanceId.hashCode()
    result = 31 * result + (queueId?.hashCode() ?: 0)
    return result
}
```

if you then go on to make a `MutableSet` of `Message`, it should insert/remove correctly.

Reply:
The `instanceId` field is generated on object instantiation meaning that when the queue is fetched and a list of messages is created locally, they will always have a different `instanceId`.

Now, if we remove it from comparison in the `equals` method, it would work in our local store *set* case but it would make it useless in every other case since `instanceId` would make a lot of sense to be included in the comparison for the `Message` class.

Let me know if you need any further clarifications.

Reply:
If we were to change the type, I think a Map would make more sense in this case, what do you think?